### ISSUE TITLE
Add RCON + Console

### ISSUE THREAD

**Issue by Kumpelinus** [hyperion-mc/hyperion#700]
Add a way to connect via RCON and trigger commands that way.

**Comment by Kumpelinus** [comment]
Already have some ideas on this Issue btw. It is on my TODO (not the highest prirority tho)

**Comment by TestingPlant** [comment]
RCON is insecure to expose publicly because it's not encrypted. IMO implementing it isn't necessary

**Comment by andrewgazelka** [comment]
> RCON is insecure to expose publicly because it's not encrypted. IMO implementing it isn't necessary

tbf could always make it bind to localhost by default. also **SHOULD** be an optimal module in a separate crate.

**Comment by Kumpelinus** [comment]
> > RCON is insecure to expose publicly because it's not encrypted. IMO implementing it isn't necessary
> 
> 
> 
> tbf could always make it bind to localhost by default. also **SHOULD** be an optional module in a separate crate.

Agreed that it shouldn't be in the core. But the main point of this is to have a way to trigger commands as a non player entity (might become useful in the future). RCON is just the usecase that is easy to implement then. It is optional tho.

**Comment by andrewgazelka** [comment]
> > > RCON is insecure to expose publicly because it's not encrypted. IMO implementing it isn't necessary
> > 
> > 
> > tbf could always make it bind to localhost by default. also **SHOULD** be an optional module in a separate crate.
> 
> Agreed that it shouldn't be in the core. But the main point of this is to have a way to trigger commands as a non player entity (might become useful in the future). RCON is just the usecase that is easy to implement then. It is optional tho.

ok easy to impl but low prio... closing this for now so it is not clogging issue board. btw closing as not planned but going to go back after event and re-open a lot of these

**Comment by andrewgazelka** [comment]
Warren Buffett is often associated with prioritizing tasks through the “Rule of 5,” which emphasizes focusing on what truly matters. The saying you’re referring to might align with this framework:

“The difference between successful people and really successful people is that really successful people say no to almost everything.”

This philosophy encourages individuals to identify their top priorities and avoid distractions or tasks that don’t contribute significantly to their primary goals. Buffett advises making a list of your top 25 goals, focusing only on the top 5, and avoiding the other 20 “like the plague,” as they can divert your attention and resources from what truly matters.

**Comment by Kumpelinus** [comment]
> > > > RCON is insecure to expose publicly because it's not encrypted. IMO implementing it isn't necessary
> 
> > > 
> 
> > > 
> 
> > > tbf could always make it bind to localhost by default. also **SHOULD** be an optional module in a separate crate.
> 
> > 
> 
> > Agreed that it shouldn't be in the core. But the main point of this is to have a way to trigger commands as a non player entity (might become useful in the future). RCON is just the usecase that is easy to implement then. It is optional tho.
> 
> 
> 
> ok easy to impl but low prio... closing this for now so it is not clogging issue board. btw closing as not planned but going to go back after event and re-open a lot of these

I wanted to do the same thing right now lol
I just wanted to have a tracking Issue for it (in case I need a task to work on when I don't have enough brain power for something more difficult)
